src/Uno.UI/UI/Xaml/Controls/VisualStatesImplementation.tt

```diff
@@ -167,46 +195,81 @@ namespace <#= mixin.NamespaceName #>
 		}
 #endif
 
-        private string GetCombinedStatePrefix() {
+#if <#= mixin.HasCommonCheckedState #> || <#= mixin.HasCombinedCheckedState #>
+		private string GetFullState()
+		{
+			var commonState = GetCommonState();
+			var checkedState = GetCheckedState();
+
 #if <#= mixin.HasCommonCheckedState #>
-            return (this.IsChecked != null && this.IsChecked.Value) ?
-                __checkedState :
-                "";
-#elif <#= mixin.HasCombinedCheckedState #>
-            return (this.IsChecked != null && this.IsChecked.Value) ?
-                __checkedState :
-                __uncheckedState;
-#else
-                return "";
+			if (checkedState == __uncheckedState)
+			{
+				return commonState;
+			}
+			else if (commonState == __normalState)
+			{
+				return checkedState;
+			}
+#endif
+			return checkedState + commonState;
+		}
 #endif
-        }
 
-        private string GetCombinedStateSuffix() {
+		private string GetCommonState()
+		{
 			var targetState = __normalState;
 			if (!this.IsEnabled)
 			{
-				targetState = __disabledState;
+				return __disabledState;
 			}
 #if <#= mixin.HasCommonPressedState #>
 			else if (IsPointerPressed && IsPointerOver)
 			{
-				targetState = __pressedState;
+				return __pressedState;
+			}
+#endif
+#if <#= mixin.HasCommonOverState #>
+			else if (IsPointerOver)
+			{
+				return __pointerOverState;
 			}
 #endif
 #if <#= mixin.HasCommonFocusedState #>
 			else if (this.IsFocused)
 			{
-				targetState = __focusedState;
+				return __focusedState;
 			}
 #endif
 
-#if !<#= mixin.HasCombinedCheckedState #>
-            if (targetState == __normalState) {
-                return "";
-            }
+// #if !<#= mixin.HasCombinedCheckedState #>
```

Comment:
Not used anymore ?

Reply:
Yes it now done in `GetFullState()` , will remove dead code.